src/table/components/head/styles.ts

```diff
@@ -21,15 +21,16 @@ export const StyledHeadRow = styled(TableRow, {
 }));
 
 export const StyledHeadCell = styled(TableCell, {
-  shouldForwardProp: (prop: string) => prop !== 'headerStyle',
-})(({ headerStyle }) => ({
+  shouldForwardProp: (prop: string) => prop !== 'headerStyle' && prop !== 'isInteractionEnabled',
+})(({ headerStyle, isInteractionEnabled }) => ({
   ...COMMON_CELL_STYLING,
   ...headerStyle,
   lineHeight: '150%',
   pointer: 'cursor',
   borderWidth: '1px 1px 1px 0px',
   '&&:hover, &&:focus': {
-    '& svg, & button': { opacity: 1 },
+    '& button': { opacity: 1 },
+    '& svg': { opacity: !isInteractionEnabled ? 0 : 1 },
```

Comment:
There is no reason to use the logical NOT operator here
```suggestion
    '& svg': { opacity: isInteractionEnabled ? 1 : 0 },
```

Reply:
Hiding arrow icon when it is in selection mode, but it hides the button as well which I don't want :(